cli/internal/cmd/init_test.go

```diff
@@ -103,31 +104,36 @@ func TestInitialize(t *testing.T) {
 		"initialize some gcp instances": {
 			provider:      cloudprovider.GCP,
 			idFile:        &clusterid.File{IP: "192.0.2.1"},
+			stateFile:     &state.State{Version: state.Version1, Infrastructure: state.Infrastructure{ClusterEndpoint: "192.0.2.1"}},
```

Comment:
nit: should we use the constructor and make the fields private if we want to go with the struct approach for stronger type safety? This would also allow to do some validation if we want to later on

Reply:
We won't be able to marshal it to YAML then without the hacks Otto mentioned. I don't know if we want to commit to this yet. I think using constructors in non-test code is a great idea. But here, IMO, it makes it less readable, as you know what's inside the struct directly.

Reply:
oh you are right! Yeah lets keep as is then